src/main/java/org/tron/core/capsule/AccountCapsule.java

```diff
@@ -296,6 +429,40 @@ public boolean addAssetAmount(byte[] key, long amount) {
     return true;
   }
 
+  /**
+   * add asset amount.
+   */
+  public boolean addAssetAmountV2(byte[] key, long amount, Manager manager) {
+    if (manager.getDynamicPropertiesStore().getAllowSameTokenName() == 0) {
+      byte[] tokenName = key;
```

Comment:
![Vulnerability](https://sonarsource.github.io/sonar-github/vulnerability.png 'Vulnerability') Vulnerability: Store a copy of "key". ([squid:S2384](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS2384))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=1647&issues=AWcW2bxfgA_rbE7wDiYW&open=AWcW2bxfgA_rbE7wDiYW)

[ISSUE_KEY]: AWcW2bxfgA_rbE7wDiYW